### PR TITLE
chore: upgrade to gradle wrapper to 7.2

### PR DESCRIPTION
**Pull request by lionel-nj**
Relates to #734 

**Summary:**

This PR upgrades the gradle wrapper to version 7.2. This is a required step for #933.

**Expected behavior:** 

No code change, application should be packaged as usual. Same tests should pass. 

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `gradle test` to make sure you didn't break anything
- [x] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [x] Linked all relevant issues
- ~[ ] Include screenshot(s) showing how this pull request works and fixes the issue(s)~

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,6 +1,5 @@
-#Thu Nov 21 11:46:52 EST 2019
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.3-all.zip
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-zipStorePath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.2-bin.zip
```
Comment (barbeau) [inline gradle/wrapper/gradle-wrapper.properties:3]: Do you want `all` here again? Looks like we had that previously.

**Review by barbeau — COMMENTED** [review]
Thanks @lionel-nj. Could you please rebase this on main branch so that the GitHub workflows to verify the Gradle wrapper run?

File: gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
Comment (lionel-nj) [reply gradle/wrapper/gradle-wrapper.properties]: Indeed

**Comment by lionel-nj** [comment]
Upgrade instructions 
```
./gradlew wrapper --gradle-version=7.2
```

**Comment by barbeau** [comment]
If you want to use `all`, I'd suggest running:

```
./gradlew wrapper --gradle-version=7.2 --distribution-type=all
```

...and committing the result of that. It might be the same as just changing `bin` to `all`, but I'm not sure.

**Comment by lionel-nj** [comment]
@barbeau  PTAL

**Review by maximearmstrong — APPROVED** [review]
LGTM :)

**Comment by lionel-nj** [comment]
Thanks for review @maximearmstrong @barbeau!

**Pull request merged**